pkg/storegateway/indexheader/index/postings.go

```diff
@@ -418,13 +471,18 @@ func (t *PostingOffsetTableV2) LabelValues(name string, prefix string, filter fu
 		return nil, nil
 	}
 
+	if filter == nil {
+		filter = func(string) bool { return true }
+	}
```

Comment:
I'd revert this to the previous `filter == nil || filter(...)` condition, as this is an extra function call which can be avoided in many cases.